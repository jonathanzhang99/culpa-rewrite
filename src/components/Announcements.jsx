import PropTypes from "prop-types";
import React from "react";

import ErrorComponent from "components/common/ErrorComponent";
import LoadingComponent from "components/common/LoadingComponent";
import useDataFetch from "components/common/useDataFetch";

const propTypes = {
  messages: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export function AnnouncementsSection(props) {
  // TODO: Implement Announcements
  const { messages } = props;
  return <div>{messages}</div>;
}

AnnouncementsSection.propTypes = propTypes;

export default function Announcements() {
  const {
    data: { messages },
    isLoading,
    isError,
  } = useDataFetch("/api/announcements/all", {
    messages: [],
  });

  if (isLoading || isError) {
    return isLoading ? <LoadingComponent /> : <ErrorComponent />;
  }

  return <AnnouncementsSection messages={messages} />;
}
