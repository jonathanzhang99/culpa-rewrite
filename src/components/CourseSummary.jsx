import PropTypes from "prop-types";
import React, { useState } from "react";
import { Icon, Accordion, Table, Grid, Container } from "semantic-ui-react";

import { CourseDisplayName } from "components/common/CourseDisplay";
import CreateReviewButton from "components/common/CreateReviewButton";
import { ProfessorDisplayLink } from "components/common/ProfessorDisplay";
import ReviewCard from "components/reviews/ReviewCard";

const reviewPropType = PropTypes.shape({
  reviewType: PropTypes.string.isRequired,
  reviewHeader: PropTypes.shape({
    courseId: PropTypes.number.isRequired,
    courseName: PropTypes.string.isRequired,
    courseCode: PropTypes.string.isRequired,
  }).isRequired,
  votes: PropTypes.shape({
    initUpvoteCount: PropTypes.number.isRequired,
    initDownvoteCount: PropTypes.number.isRequired,
    initFunnyCount: PropTypes.number.isRequired,
    upvoteClicked: PropTypes.bool.isRequired,
    downvoteClicked: PropTypes.bool.isRequired,
    funnyClicked: PropTypes.bool.isRequired,
  }).isRequired,
  workload: PropTypes.string,
  submissionDate: PropTypes.string.isRequired,
  reviewId: PropTypes.number.isRequired,
  deprecated: PropTypes.bool,
  content: PropTypes.string,
});

const defaultReview = {
  reviewType: "course",
  reviewHeader: {
    courseId: 0,
    courseName: "Course Name",
    courseCode: "Course Code",
  },
  votes: {
    initUpvoteCount: 0,
    initDownvoteCount: 0,
    initFunnyCount: 0,
    upvoteClicked: false,
    downvoteClicked: false,
    funnyClicked: false,
  },
  workload: "",
  submissionDate: "",
  reviewId: 0,
  deprecated: false,
  content: "This is a review.",
};

const propTypesCourseSummary = {
  courseId: PropTypes.string.isRequired,
  courseSummary: PropTypes.shape({
    courseName: PropTypes.string.isRequired,
    courseCallNumber: PropTypes.string.isRequired,
    departmentName: PropTypes.string.isRequired,
    associatedProfessors: PropTypes.array.isRequired,
  }).isRequired,
  reviewSummary: PropTypes.shape({
    positiveReview: reviewPropType,
    negativeReview: reviewPropType,
  }),
};

const defaultPropsCourseSummary = {
  reviewSummary: {
    positiveReivew: defaultReview,
    negativeReview: defaultReview,
  },
};

export default function CourseSummary({
  courseId,
  courseSummary,
  reviewSummary,
}) {
  return (
    <Grid>
      <Grid.Row>
        <CourseHeader courseId={courseId} courseSummary={courseSummary} />
      </Grid.Row>
      <Grid.Row>
        <ReviewSummary reviewSummary={reviewSummary} />
      </Grid.Row>
    </Grid>
  );
}

CourseSummary.propTypes = propTypesCourseSummary;
CourseSummary.defaultProps = defaultPropsCourseSummary;

const propTypesCourseHeader = {
  courseId: PropTypes.string.isRequired,
  courseSummary: PropTypes.shape({
    courseName: PropTypes.string.isRequired,
    courseCallNumber: PropTypes.string.isRequired,
    departmentName: PropTypes.string.isRequired,
    associatedProfessors: PropTypes.array.isRequired,
  }).isRequired,
};

export function CourseHeader({ courseId, courseSummary }) {
  const [isAccordionActive, setAccordionActive] = useState(false);
  const {
    courseName,
    courseCallNumber,
    departmentName,
    associatedProfessors,
  } = courseSummary;

  const displayAccordion = associatedProfessors.length > 5;

  const ProfessorAccordion = (
    <Accordion>
      <Accordion.Title
        active={isAccordionActive}
        onClick={() => setAccordionActive(!isAccordionActive)}
      >
        <Icon name={!isAccordionActive ? "angle down" : "angle up"} />
        {!isAccordionActive ? "Show" : "Hide"} all professors who teach this
        course
      </Accordion.Title>
      <Accordion.Content active={isAccordionActive}>
        <Table basic="very" textAlign="left">
          <tbody>
            {/* this is to prevent bugs from browser inserting <tbody> */}
            {associatedProfessors.map((professor) => {
              const {
                firstName,
                lastName,
                professorId,
                profDepartments,
              } = professor;
              return (
                <Table.Row key={professorId}>
                  <Table.Cell key={professorId}>
                    <ProfessorDisplayLink
                      firstName={firstName}
                      lastName={lastName}
                      professorId={professorId}
                    />
                  </Table.Cell>
                  <Table.Cell key={`${professorId}_departments`}>
                    {profDepartments.map((department, index) => {
                      const {
                        profDepartmentId,
                        profDepartmentName,
                      } = department;
                      return (
                        <span key={profDepartmentId}>
                          {profDepartmentId}: {profDepartmentName}
                          {profDepartments.length - 1 !== index ? ", " : ""}
                        </span>
                      );
                    })}
                  </Table.Cell>
                </Table.Row>
              );
            })}
          </tbody>
        </Table>
      </Accordion.Content>
    </Accordion>
  );

  const ProfessorList = (
    <div>
      Professors:{" "}
      {associatedProfessors.map((professor, index) => {
        const { firstName, lastName, professorId } = professor;
        return (
          <span key={`${lastName}_${professorId}`}>
            <ProfessorDisplayLink
              firstName={firstName}
              lastName={lastName}
              professorId={professorId}
            />
            {associatedProfessors.length - 1 !== index ? ", " : ""}
          </span>
        );
      })}
    </div>
  );

  // TODO: Integrate DepartmentDisplay
  return (
    <Container>
      <CourseDisplayName
        as="header"
        courseCallNumber={courseCallNumber}
        courseName={courseName}
      />
      <div> Department: {departmentName} </div>

      {displayAccordion ? ProfessorAccordion : ProfessorList}

      <CreateReviewButton compact color="yellow" courseId={courseId.toString()}>
        WRITE A REVIEW FOR {courseName}
      </CreateReviewButton>
    </Container>
  );
}

CourseHeader.propTypes = propTypesCourseHeader;

const propTypesReviewSummary = {
  reviewSummary: PropTypes.oneOfType([
    PropTypes.shape({
      positiveReview: reviewPropType,
      negativeReview: reviewPropType,
    }),
    PropTypes.shape({
      mostAgreedReview: reviewPropType,
    }),
  ]),
};

const defaultPropsReviewSummary = {
  reviewSummary: {
    positiveReivew: defaultReview,
    negativeReview: defaultReview,
  },
};

function ReviewSummary({ reviewSummary }) {
  if ("mostAgreedReview" in reviewSummary) {
    const { mostAgreedReview } = reviewSummary;
    return (
      <Container>
        <Grid>
          <Grid.Row>
            <Grid.Column width={14}>
              <h3>Most Agreed Review</h3>
              <CourseReviewCard review={mostAgreedReview} />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Container>
    );
  }

  if ("positiveReview" in reviewSummary && "negativeReview" in reviewSummary) {
    const { positiveReview, negativeReview } = reviewSummary;
    return (
      <Container>
        <Grid>
          <Grid.Row>
            <Grid.Column width={7}>
              <h3>Most Positive Review</h3>
              <CourseReviewCard review={positiveReview} />
            </Grid.Column>
            <Grid.Column width={1} />
            <Grid.Column width={7}>
              <h3>Most Negative Review</h3>
              <CourseReviewCard review={negativeReview} />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Container>
    );
  }

  return null;
}

ReviewSummary.propTypes = propTypesReviewSummary;
ReviewSummary.defaultProps = defaultPropsReviewSummary;

const propTypesCourseReviewCard = {
  review: reviewPropType.isRequired,
};

function CourseReviewCard({ review }) {
  const {
    reviewType,
    reviewHeader,
    votes,
    workload,
    submissionDate,
    reviewId,
    deprecated,
    content,
  } = review;

  return (
    <ReviewCard
      content={content}
      deprecated={deprecated}
      reviewHeader={reviewHeader}
      reviewId={reviewId}
      reviewType={reviewType}
      submissionDate={submissionDate}
      votes={votes}
      workload={workload}
    />
  );
}

CourseReviewCard.propTypes = propTypesCourseReviewCard;
