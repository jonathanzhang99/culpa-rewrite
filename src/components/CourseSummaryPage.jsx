import PropTypes from "prop-types";
import React from "react";
import { useParams } from "react-router-dom";

import { CourseDisplayName } from "components/common/CourseDisplay";
import CreateReviewButton from "components/common/CreateReviewButton";
import ErrorComponent from "components/common/ErrorComponent";
import LoadingComponent from "components/common/LoadingComponent";
import { ProfessorDisplayLink } from "components/common/ProfessorDisplay";
import useDataFetch from "components/common/useDataFetch";

const propTypesCourseHeader = {
  courseId: PropTypes.number.isRequired,
  courseSummary: PropTypes.shape({
    courseName: PropTypes.string.isRequired,
    departmentName: PropTypes.string.isRequired,
    recentCourseInstances: PropTypes.array.isRequired,
    associatedProfessors: PropTypes.array.isRequired,
  }).isRequired,
};

export default function CourseSummary() {
  const { courseId } = useParams();

  const {
    data: { courseSummary },
    isLoading,
    isError,
  } = useDataFetch(`/api/courses/${courseId}`, {
    courseSummary: {
      courseName: "",
      departmentName: "",
      recentCourseInstances: [
        {
          year: 0,
          semester: 0,
        },
      ],
      associatedProfessors: [
        {
          firstName: "",
          lastName: "",
          professorId: 0,
        },
      ],
    },
  });

  if (isLoading) {
    return <LoadingComponent />;
  }

  if (isError) {
    return <ErrorComponent />;
  }

  return (
    <div>
      <CourseHeader courseId={courseId} courseSummary={courseSummary} />
      <ReviewSummary />
    </div>
  );
}

function CourseHeader({ courseId, courseSummary }) {
  const {
    courseName,
    departmentName,
    recentCourseInstances,
    associatedProfessors,
  } = courseSummary;

  return (
    <div>
      <h1>
        <CourseDisplayName code="COURSE_CODE" name={courseName} />
      </h1>
      <h3>{departmentName}</h3>

      <h3>
        Professors:{" "}
        {associatedProfessors.map((professor, index) => {
          const { firstName, lastName, professorId } = professor;
          return (
            <span>
              <ProfessorDisplayLink
                as="a"
                firstName={firstName}
                lastName={lastName}
                professorId={professorId}
              />
              {associatedProfessors.length - 1 !== index ? ", " : ""}
            </span>
          );
        })}
      </h3>

      <h3>
        Offered in:{" "}
        {recentCourseInstances.map((courseInstance, index) => {
          const { year, semester } = courseInstance;
          return (
            <span>
              {year} {semester}{" "}
              {recentCourseInstances.length - 1 !== index ? ", " : ""}
            </span>
          );
        })}
      </h3>

      <CreateReviewButton compact color="yellow" courseId={courseId}>
        Write a Review for {courseName}
      </CreateReviewButton>
    </div>
  );
}

function ReviewSummary() {
  return "Review Summary here";
}

CourseHeader.propTypes = propTypesCourseHeader;
