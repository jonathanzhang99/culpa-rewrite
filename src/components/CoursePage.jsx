import React from "react";
import { useParams } from "react-router-dom";

import ErrorComponent from "components/common/ErrorComponent";
import LoadingComponent from "components/common/LoadingComponent";
import useDataFetch from "components/common/useDataFetch";
import CourseSummary from "components/CourseSummary";
import ReviewSection from "components/reviews/ReviewSection";

export default function CoursePage() {
  const pageType = "course";
  const { courseId } = useParams();

  const defaultReview = {
    reviewType: "course",
    reviewHeader: {
      profId: 0,
      profFirstName: "First",
      profLastName: "Last",
    },
    votes: {
      initUpvoteCount: 0,
      initDownvoteCount: 0,
      initFunnyCount: 0,
      upvoteClicked: false,
      downvoteClicked: false,
      funnyClicked: false,
    },
    workload: "",
    submissionDate: "",
    reviewId: 0,
    deprecated: false,
    content: "This is a review.",
  };

  const courseDataFetched = useDataFetch(`/api/course/${courseId}`, {
    courseSummary: {
      courseName: "",
      courseCallNumber: "",
      departmentId: 0,
      departmentName: "",
      associatedProfessors: [],
    },
    review_summary: {
      positiveReview: defaultReview,
      negativeReview: defaultReview,
    },
  });

  const { courseSummary, reviewSummary } = courseDataFetched.data;
  const isCourseLoading = courseDataFetched.isLoading;
  const isCourseError = courseDataFetched.isError;

  const reviewDataFetched = useDataFetch(`/api/review/get/course/${courseId}`, {
    reviews: [],
  });

  const { reviews } = reviewDataFetched.data;
  const isReviewLoading = reviewDataFetched.isLoading;
  const isReviewError = reviewDataFetched.isError;

  if (isCourseLoading || isCourseError) {
    return isCourseLoading ? <LoadingComponent /> : <ErrorComponent />;
  }

  if (isReviewLoading || isReviewError) {
    return isReviewLoading ? <LoadingComponent /> : <ErrorComponent />;
  }

  return (
    <>
      <CourseSummary
        courseId={Number(courseId)}
        courseSummary={courseSummary}
        reviewSummary={reviewSummary}
      />
      <ReviewSection
        associatedEntities={courseSummary.associatedProfessors}
        id={Number(courseId)}
        initReviews={reviews}
        pageType={pageType}
      />
    </>
  );
}
