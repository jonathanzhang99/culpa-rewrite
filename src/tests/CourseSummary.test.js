import { render, fireEvent, screen, cleanup } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { CourseHeader } from "components/CourseSummary";

afterEach(cleanup);

describe("CourseSummary Components", () => {
  const courseHeaderTestCases = [
    {
      testName: "empty props CourseSummary",
      courseId: "0",
      courseSummary: {
        courseName: "",
        courseCallNumber: "",
        departmentName: "",
        associatedProfessors: [
          {
            firstName: "",
            lastName: "",
            professorId: 0,
            profDepartments: [
              {
                profDepartmentId: 0,
                profDepartmentName: "",
              },
            ],
          },
        ],
      },
    },
    {
      testName: "empty professors CourseSummary",
      courseId: "0",
      courseSummary: {
        courseName: "Machine Learning",
        courseCallNumber: "COMS 4771",
        departmentName: "Computer Science",
        associatedProfessors: [
          {
            firstName: "",
            lastName: "",
            professorId: 0,
            profDepartments: [
              {
                profDepartmentId: 0,
                profDepartmentName: "",
              },
            ],
          },
        ],
      },
    },
    {
      testName: "Shows one professor in a list",
      courseId: "1",
      courseSummary: {
        courseName: "Machine Learning",
        courseCallNumber: "COMS 4771",
        departmentName: "Computer Science",
        associatedProfessors: [
          {
            firstName: "Nakul",
            lastName: "Verma",
            professorId: 1,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
        ],
      },
    },
    {
      testName: "Shows six professors in a list",
      courseId: "1",
      courseSummary: {
        courseName: "Machine Learning",
        courseCallNumber: "COMS 4771",
        departmentName: "Computer Science",
        associatedProfessors: [
          {
            firstName: "Professor",
            lastName: "One",
            professorId: 1,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
          {
            firstName: "Professor",
            lastName: "Two",
            professorId: 2,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
          {
            firstName: "Professor",
            lastName: "Three",
            professorId: 3,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
          {
            firstName: "Professor",
            lastName: "Four",
            professorId: 4,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
          {
            firstName: "Professor",
            lastName: "Five",
            professorId: 5,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
          {
            firstName: "Professor",
            lastName: "Six",
            professorId: 6,
            profDepartments: [
              {
                profDepartmentId: 1,
                profDepartmentName: "Computer Science",
              },
              {
                profDepartmentId: 2,
                profDepartmentName: "Law",
              },
            ],
          },
        ],
      },
    },
  ];

  courseHeaderTestCases.forEach(({ testName, courseId, courseSummary }) => {
    test(testName, () => {
      const snapshot = render(
        <MemoryRouter>
          <CourseHeader courseId={courseId} courseSummary={courseSummary} />
        </MemoryRouter>
      );
      expect(snapshot).toMatchSnapshot();
    });
  });

  /* semantic-ui react does not change html for folding/unfolding so 
   checking for button string change instead */
  test("Shows six professors in an accordion", () => {
    const courseId = "1";
    const courseSummary = {
      courseName: "Machine Learning",
      courseCallNumber: "COMS 4771",
      departmentName: "Computer Science",
      associatedProfessors: [
        {
          firstName: "Professor",
          lastName: "One",
          professorId: 1,
          profDepartments: [
            {
              profDepartmentId: 1,
              profDepartmentName: "Computer Science",
            },
            {
              profDepartmentId: 2,
              profDepartmentName: "Law",
            },
          ],
        },
        {
          firstName: "Professor",
          lastName: "Two",
          professorId: 2,
          profDepartments: [
            {
              profDepartmentId: 1,
              profDepartmentName: "Computer Science",
            },
            {
              profDepartmentId: 2,
              profDepartmentName: "Law",
            },
          ],
        },
        {
          firstName: "Professor",
          lastName: "Three",
          professorId: 3,
          profDepartments: [
            {
              profDepartmentId: 1,
              profDepartmentName: "Computer Science",
            },
            {
              profDepartmentId: 2,
              profDepartmentName: "Law",
            },
          ],
        },
        {
          firstName: "Professor",
          lastName: "Four",
          professorId: 4,
          profDepartments: [
            {
              profDepartmentId: 1,
              profDepartmentName: "Computer Science",
            },
            {
              profDepartmentId: 2,
              profDepartmentName: "Law",
            },
          ],
        },
        {
          firstName: "Professor",
          lastName: "Five",
          professorId: 5,
          profDepartments: [
            {
              profDepartmentId: 1,
              profDepartmentName: "Computer Science",
            },
            {
              profDepartmentId: 2,
              profDepartmentName: "Law",
            },
          ],
        },
        {
          firstName: "Professor",
          lastName: "Six",
          professorId: 6,
          profDepartments: [
            {
              profDepartmentId: 1,
              profDepartmentName: "Computer Science",
            },
            {
              profDepartmentId: 2,
              profDepartmentName: "Law",
            },
          ],
        },
      ],
    };

    render(
      <MemoryRouter>
        <CourseHeader courseId={courseId} courseSummary={courseSummary} />
      </MemoryRouter>
    );
    expect(
      screen.queryByText("Show all professors who teach this course")
    ).toBeInTheDocument();

    fireEvent.click(
      screen.getByText("Show all professors who teach this course")
    );
    expect(
      screen.queryByText("Hide all professors who teach this course")
    ).toBeInTheDocument();

    fireEvent.click(
      screen.getByText("Hide all professors who teach this course")
    );
    expect(
      screen.queryByText("Show all professors who teach this course")
    ).toBeInTheDocument();
  });
});
