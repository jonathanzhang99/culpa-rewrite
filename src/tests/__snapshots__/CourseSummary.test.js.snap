// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseSummary Components empty professors CourseSummary 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          <p
            class="ui header"
          >
            COMS4771
             
            Machine Learning
          </p>
        </h1>
        <h3>
          Department: 
          Computer Science
        </h3>
        <div
          class="accordion ui"
        >
          <h3
            class="title"
          >
            <i
              aria-hidden="true"
              class="angle down icon"
            />
            Show
             all professors who teach this course
          </h3>
          <div
            class="content"
          >
            <table
              class="ui very basic left aligned table"
            >
              <tbody>
                <tr
                  class=""
                >
                  <td
                    class=""
                  >
                    <a
                      href="/professors/0"
                    >
                      <h5
                        class="ui header"
                      >
                        
                         
                        
                      </h5>
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      0
                      : 
                      
                      
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <a
          href="/review"
        >
          <button
            class="ui yellow compact button"
          >
            WRITE A REVIEW FOR 
            Machine Learning
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        <p
          class="ui header"
        >
          COMS4771
           
          Machine Learning
        </p>
      </h1>
      <h3>
        Department: 
        Computer Science
      </h3>
      <div
        class="accordion ui"
      >
        <h3
          class="title"
        >
          <i
            aria-hidden="true"
            class="angle down icon"
          />
          Show
           all professors who teach this course
        </h3>
        <div
          class="content"
        >
          <table
            class="ui very basic left aligned table"
          >
            <tbody>
              <tr
                class=""
              >
                <td
                  class=""
                >
                  <a
                    href="/professors/0"
                  >
                    <h5
                      class="ui header"
                    >
                      
                       
                      
                    </h5>
                  </a>
                </td>
                <td
                  class=""
                >
                  <span>
                    0
                    : 
                    
                    
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <a
        href="/review"
      >
        <button
          class="ui yellow compact button"
        >
          WRITE A REVIEW FOR 
          Machine Learning
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CourseSummary Components empty props CourseSummary 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          <p
            class="ui header"
          >
            
             
            
          </p>
        </h1>
        <h3>
          Department: 
          
        </h3>
        <div
          class="accordion ui"
        >
          <h3
            class="title"
          >
            <i
              aria-hidden="true"
              class="angle down icon"
            />
            Show
             all professors who teach this course
          </h3>
          <div
            class="content"
          >
            <table
              class="ui very basic left aligned table"
            >
              <tbody>
                <tr
                  class=""
                >
                  <td
                    class=""
                  >
                    <a
                      href="/professors/0"
                    >
                      <h5
                        class="ui header"
                      >
                        
                         
                        
                      </h5>
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      0
                      : 
                      
                      
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <a
          href="/review"
        >
          <button
            class="ui yellow compact button"
          >
            WRITE A REVIEW FOR 
            
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        <p
          class="ui header"
        >
          
           
          
        </p>
      </h1>
      <h3>
        Department: 
        
      </h3>
      <div
        class="accordion ui"
      >
        <h3
          class="title"
        >
          <i
            aria-hidden="true"
            class="angle down icon"
          />
          Show
           all professors who teach this course
        </h3>
        <div
          class="content"
        >
          <table
            class="ui very basic left aligned table"
          >
            <tbody>
              <tr
                class=""
              >
                <td
                  class=""
                >
                  <a
                    href="/professors/0"
                  >
                    <h5
                      class="ui header"
                    >
                      
                       
                      
                    </h5>
                  </a>
                </td>
                <td
                  class=""
                >
                  <span>
                    0
                    : 
                    
                    
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <a
        href="/review"
      >
        <button
          class="ui yellow compact button"
        >
          WRITE A REVIEW FOR 
          
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CourseSummary Components valid props CourseSummary 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          <p
            class="ui header"
          >
            COMS4771
             
            Machine Learning
          </p>
        </h1>
        <h3>
          Department: 
          Computer Science
        </h3>
        <div
          class="accordion ui"
        >
          <h3
            class="title"
          >
            <i
              aria-hidden="true"
              class="angle down icon"
            />
            Show
             all professors who teach this course
          </h3>
          <div
            class="content"
          >
            <table
              class="ui very basic left aligned table"
            >
              <tbody>
                <tr
                  class=""
                >
                  <td
                    class=""
                  >
                    <a
                      href="/professors/1"
                    >
                      <h5
                        class="ui header"
                      >
                        Nakul
                         
                        Verma
                      </h5>
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      1
                      : 
                      Computer Science
                      , 
                    </span>
                    <span>
                      2
                      : 
                      Law
                      
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <a
          href="/review"
        >
          <button
            class="ui yellow compact button"
          >
            WRITE A REVIEW FOR 
            Machine Learning
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        <p
          class="ui header"
        >
          COMS4771
           
          Machine Learning
        </p>
      </h1>
      <h3>
        Department: 
        Computer Science
      </h3>
      <div
        class="accordion ui"
      >
        <h3
          class="title"
        >
          <i
            aria-hidden="true"
            class="angle down icon"
          />
          Show
           all professors who teach this course
        </h3>
        <div
          class="content"
        >
          <table
            class="ui very basic left aligned table"
          >
            <tbody>
              <tr
                class=""
              >
                <td
                  class=""
                >
                  <a
                    href="/professors/1"
                  >
                    <h5
                      class="ui header"
                    >
                      Nakul
                       
                      Verma
                    </h5>
                  </a>
                </td>
                <td
                  class=""
                >
                  <span>
                    1
                    : 
                    Computer Science
                    , 
                  </span>
                  <span>
                    2
                    : 
                    Law
                    
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <a
        href="/review"
      >
        <button
          class="ui yellow compact button"
        >
          WRITE A REVIEW FOR 
          Machine Learning
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
